server/raft.go

```diff
@@ -3619,6 +3613,17 @@ func (n *raft) processAppendEntryResponse(ar *appendEntryResponse) {
 		n.stepdownLocked(noLeader)
 		n.Unlock()
 		arPool.Put(ar)
+		return
+	} else if ar.term < n.term {
```

Comment:
Don't need this else, just normal if here since you return from the previous block.